maggy/core/environment_singleton.py

```diff
@@ -0,0 +1,28 @@
+import os
+
+"""
+return an instance of the environment to be used by maggy within a session.
+"""
+
+
+def environment_singleton():
+    global environmentInstance
```

Comment:
I don't have a strong opinion on which way we go to implement the singleton. I just think it should follow the pattern of having a `get_instance()` so this will make it clear that it's a singleton and also we don't have to call it `singleton`. This is how I would have done it: https://github.com/logicalclocks/feature-store-api/blob/master/python/hsfs/client/__init__.py

I haven't used `globals()` before and I don't know everything about its internals. But does it differ from declaring a variable as `global`?